### PR TITLE
Enhance loading bars component with additional (optional) text

### LINKED ISSUE #51: Enchance loading bars component with additional (optional) text below icon
## Description

The loading bars component which displays a loading "spinner", which symbolises data or structure loading, should get a optional loading text `title` and `description` field. 

- Package: course 

## Expected Behavior

When using the shared loading bars component I want to be able to display a loading title (with `...` eg. `Loading course...`) and provide it with input or manual text to the shared component. **Optional and by default hidden* 

## Actual Behavior

You can only show the loading bars gif icon, without additional information.

### PR DESCRIPTION
**Pull request by MRX1621**
## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** and other relevant documents.

## Description

fix(course): implement "Is loading..." - Message and place loading-icon in centre

## Screenshots (if appropriate):
image

### CODE REVIEW

**Review by danieljancar — CHANGES_REQUESTED** [review]
I only want the changes in the `loading-bars` component and not the loading component being imported and used in the `agenda` and `home` components, which you probably used to test your loading additions.

File: apps/course/src/app/shared/feedback/loading-bars/loading-bars.component.html [hunk 1]
```diff
@@ -9,4 +9,9 @@
     [class.loading-sm]="size === 'sm'"
     class="loading-bars loading"
   ></span>
+
+  <!-- Optionaler Lade-Text -->
```
Comment (danieljancar) [inline apps/course/src/app/shared/feedback/loading-bars/loading-bars.component.html:13]: Overkill and german comment. Please remove 🦖

File: apps/course/src/app/shared/feedback/loading-bars/loading-bars.component.ts [hunk 2]
```diff
@@ -2,13 +2,19 @@ import { Component, Input } from '@angular/core';
...
   templateUrl: './loading-bars.component.html',
-  styleUrl: './loading-bars.component.scss',
+  styleUrls: ['./loading-bars.component.scss'],
+  standalone: true,
 })
 export class LoadingBarsComponent {
   @Input() size: string = 'lg';
   @Input() marginTop: string = '50';
   @Input() marginBottom: string = '50';
+  @Input() loadingText: string = '';
```
Comment (danieljancar) [inline apps/course/src/app/shared/feedback/loading-bars/loading-bars.component.ts:13]: Please use type:
```ts
@Input() loadingText: string | null;
```
And in the template check if `loadingText` has a string value:
```html
@if (loadingText) {
  <p>{{ loadingText }}</p>
} 
```
Also there's no need for the `class="loading-text"` in the html tag.

File: apps/course/src/app/features/course/agenda/course-agenda.component.html [hunk 3]
```diff
@@ -1 +1,7 @@
 <p>agenda works!</p>
+<div class="flex justify-center items-center">
+  <app-loading-bars marginTop="350" size="lg"></app-loading-bars>
+</div>
+<div class="flex justify-center items-center">
+  <h1>Content is loading... Don't worry</h1>
```
Comment (danieljancar) [inline apps/course/src/app/features/course/agenda/course-agenda.component.html:6]: Please remove all this.

File: apps/course/src/app/features/course/agenda/course-agenda.component.ts [hunk 4]
```diff
@@ -3,11 +3,12 @@ import { Subscription } from 'rxjs';
 import { Course } from '../../../../../projects/types/src/lib/course.types';
 import { CourseStateService } from '../../../core/course-state.service';
 import { Title } from '@angular/platform-browser';
+import { LoadingBarsComponent } from '../../../shared/feedback/loading-bars/loading-bars.component';
 
 @Component({
   selector: 'app-agenda',
   standalone: true,
-  imports: [],
+  imports: [LoadingBarsComponent],
```
Comment (danieljancar) [inline apps/course/src/app/features/course/agenda/course-agenda.component.ts:11]: Remove

**Review by danieljancar — CHANGES_REQUESTED** [review]

**Comment by MRX1621** [comment]
is changed

File: apps/course/src/app/features/course/agenda/course-agenda.component.ts (hunk 4 above)
Comment (danieljancar) [reply apps/course/src/app/features/course/agenda/course-agenda.component.ts:11]: Still needs to be removed as it's not used yet in this component.

File: apps/course/src/app/shared/feedback/loading-bars/loading-bars.component.ts [hunk 5]
```diff
@@ -2,13 +2,19 @@ import { Component, Input } from '@angular/core';
...
 })
 export class LoadingBarsComponent {
   @Input() size: string = 'lg';
   @Input() marginTop: string = '50';
   @Input() marginBottom: string = '50';
+  @Input() loadingText: string | null;
+
+  constructor() {}
+
+  shouldShowLoadingText(): boolean {
```
Comment (danieljancar) [inline apps/course/src/app/shared/feedback/loading-bars/loading-bars.component.ts:17]: This method can be removed as it's not used anymore.

**Review by danieljancar — CHANGES_REQUESTED** [review]

**Comment by MRX1621** [comment]
V3: changed (again)

**Review by danieljancar — APPROVED** [review]
Looks good.

**Pull request merged**